lib/mycha_spec.coffee

```diff
@@ -0,0 +1,62 @@
+async = require 'async'
```

Comment:
That's not a valid point imho. Any code can be tested in its own language, or in Cucumber. There is Cucumber.js for testing JavaScript code. 

The distinction is that Mocha is for unit tests, and unit tests are for individual units of code. Cucumber is for integration tests, and to document end-user facing behavior. The choice depends on what this spec here tries to accomplish.

This spec is getting pretty complex, with setting up of test files etc, and becomes pretty unreadable. I think Cucumber would make this a lot more readable, and visualize the different ways of using Mycha and how it behaves in the different usage scenarios better.